api/internal/postgres/postgres.go

```diff
@@ -0,0 +1,359 @@
+package postgres
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+
+	"github.com/equinor/oneseismic/api/internal"
+)
+
+type coordinate struct {
+	X float64
+	Y float64
+}
+
+type Point coordinate
+
+type Linestring struct {
+	Coords []coordinate
+}
+
+type Polygon struct {
+	Coords []coordinate
+}
+
+type Geometry struct {
+	Point      *Point
+	Linestring *Linestring
+	Polygon    *Polygon
+}
+
```

Comment:
1a) What you can ask for is defined by the schema, so if the schema is not updated with the new field, you can’t ask for it. Note that the API fetches the full manifest, and will simply ignore unknown fields when parsing into the Manifest type. I.e. it’s “safe”, in the sense that nothing breaks, to add new fields via upload, but not update the API. The only effect is that you can’t reach the new data until we explicitly add it to the API

1b) If a field doesn’t exist on a particular manifest, it will yield null when querying for it.

Here is an example query and the response. The 'custom-field' manifest have an extra field called dummy, which is not implemented by the API. The 'missing-field' manifest doesn't have a `upload-filename` key:
```
{
  manifests(where: {or: [
    {eq: {guid: "missing-field"}}, 
    {eq: {guid: "custom-field"}}
  ] } ) {
    guid
    uploadFilename
  }
}
```  

```
{
  "data": {
    "manifests": [
      {
        "guid": "custom-field",
        "uploadFilename": "small.segy"
      },
      {
        "guid": "missing-field",
        "uploadFilename": null
      }
    ]
  }
}
```

2) Yes, when setting up a new separate database, we have to define the the schema and tables (and new users for that specific database). This is pretty easy tho, as postgres support template databases, which you can use as templates (duh) when creating new database-instances. I’ve made one for the catalogue. Once a database and it’s users are created, both ingest and the API can connect and do their thing

3) don’t know, I’ll have a look

Reply:
The core library from https://github.com/dosco/graphjin looks rather interesting actually. Poked into it a little bit, and while it would undoubtably be cool if we can off-load all these graphql -> sql mappings, I did note some potential issues:

1) The graphql schema is autogenerated from the db schema, which is kinda the point, but it does seem that we loose the ability to document it
2) It do have some support for `jsonb` column, but I'm unsure as to what extent. We would likely have to try it out to see. The `jsonb` comes with a caveat tho, in order for it to work, this library have to create virtual tables of the fields in the json. Shouldn't be problem w.r.t. to memory/storage, as the manifests are so small anyway. But it means that the Postgres users that the API uses will have to be granted more access than purely SELECT. Might be non-issue, don't know at this point
3) This is the big killer: there doesn't seem to be support for PostGIS columns. Maybe there is way to get around it and do that part manually, not sure 


I think that this is out-of-scope of this PR, because we'd have to prototype it to figure out if it's useful or not. But we should consider doing just that at some point, and possible port what we have if it turns out useful. The library is rather young, and under heavy development so it's not unlikely that stuff like schema docs and PostGIS support will be added in the future